legend-engine-protocol-pure/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/modelToModel/connection/ModelStoreTestConnectionFactory.java

```diff
@@ -0,0 +1,121 @@
+    //currently creates a circular dependency, hence cannot use the already present variables
+    private static String DATA_PROTOCOL_NAME = "data";
+    private static String APPLICATION_JSON = "application/json";
+    private static String APPLICATION_XML = "application/xml";
+
+    @Override
+    public Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnectionsForStore(Store store, EmbeddedData testData, List<DataElement> dataElementList)
+    {
+        if (store instanceof ModelStore && testData instanceof ModelStoreData)
+        {
+            for (Map.Entry<String, ValueSpecification> map : ((ModelStoreData) testData).instances.entrySet())
+            {
+                if (map.getValue() instanceof org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.Pair &&  ((org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.Pair) map.getValue()).second instanceof PackageableElementPtr)
+                {
+                    EmbeddedData testDataElement = Iterate.detect(dataElementList, e -> ((PackageableElementPtr)((org.finos.legend.engine.protocol.pure.v1.model.valueSpecification.raw.Pair) map.getValue()).second).fullPath.equals(e.getPath())).data;
+                    if (testDataElement instanceof ExternalFormatData && APPLICATION_JSON.equals(((ExternalFormatData) testDataElement).contentType))
+                    {
+                        JsonModelConnection jsonModelConnection = new JsonModelConnection();
+                        jsonModelConnection._class = map.getKey();
+                        jsonModelConnection.element = MODEL_STORE;
+                        return this.tryBuildTestConnection(jsonModelConnection, testDataElement);
+                    }
+                    else if (testDataElement instanceof ExternalFormatData && APPLICATION_XML.equals(((ExternalFormatData) testDataElement).contentType))
+                    {
+                        XmlModelConnection xmlModelConnection = new XmlModelConnection();
+                        xmlModelConnection._class = map.getKey();
+                        xmlModelConnection.element = MODEL_STORE;
+                        return this.tryBuildTestConnection(xmlModelConnection, testDataElement);
+                    }
+                    else
+                    {
+                        throw new RuntimeException("Data format specified is invalid, allowed types: JSON, XML for external format data");
+                    }
+                }
+                else
+                {
+                    throw new RuntimeException("Expected Model Store Data to contain Data Elements");
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Pair<Connection, List<Closeable>>> tryBuildTestConnection(Connection sourceConnection, EmbeddedData embeddedData)
+    {
+        if (sourceConnection instanceof JsonModelConnection)
```

Comment:
I believe we have similar logic in some TestConnectionVisitor. Can we reuse that

Reply:
shifted that logic here because of dependency issues

Reply:
Resolved